Src/AutoFixture.NUnit3/FixedNameTestMethodBuilder.cs

```diff
@@ -0,0 +1,84 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using System.Globalization;
+
+namespace Ploeh.AutoFixture.NUnit3
+{
+    /// <summary>
+    /// Builder that generates <see cref="TestMethod"/> with fixed names.
+    /// This is needed for some test runners such as the Nunit test adaptor for Visual Studio.
+    /// </summary>
+    /// <seealso cref="MutableNameTestMethodBuilder" />
+    public class FixedNameTestMethodBuilder : MutableNameTestMethodBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FixedNameTestMethodBuilder"/> class.
+        /// </summary>
+        public FixedNameTestMethodBuilder() { }
+
+        /// <inheritdoc />
+        protected override TestCaseParameters GetParametersForMethod(IMethodInfo method, object[] args, int autoDataStartIndex)
+        {
+            if (method == null)
+            {
+                throw new ArgumentNullException(nameof(method));
+            }
+
+            if (args == null)
+            {
+                throw new ArgumentNullException(nameof(args));
+            }
+
+            var result = base.GetParametersForMethod(method, args, autoDataStartIndex);
+            var methodParameters = method.GetParameters();
+
+            EnsureOriginalArgumentsArrayIsNotShared(result);
+
+            for (int i = autoDataStartIndex; i < result.OriginalArguments.Length; i++)
+            {
+                result.OriginalArguments[i] = new TypeNameRenderer(methodParameters[i].ParameterType);
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// In NUnit 3.5+ the Arguments and OriginalArguments properties are containing two different instances,
+        /// not in earlier versions.<BR/>
+        /// Unfortunately Arguments has a private setter and there is not way to change the instance other
```

Comment:
Fix to "OriginalArguments has a..", as you will change this property.

Reply:
> We are changing the Arguments property, not OriginalArguments.

That's the main point concern - we should modify the `OriginalArguments` array instead. It's a bit strange that we patch the `OriginalArguments` array, but make a copy of the `Arguments` one. See my comment below - it's exactly about that.

Reply:
I've just checked the source code and found that actually the setter is present, but it's private. You can use the following code snippet to set the value:
```c#
// We should use the declaring type, otherwise private setter is not available.
var property = typeof(TestParameters)
    .GetProperty(nameof(TestCaseParameters.OriginalArguments));
property.SetValue(parametersObject, newValue);
```

Currently it doesn't matter whether you access `internal` or `private` setter - code is ugly in any way 😅 However, it will be more correct to modify the `OriginalArguments` array.